build.psm1

```diff
@@ -1,62 +1,74 @@
-# Use the .NET Core APIs to determine the current platform; if a runtime
-# exception is thrown, we are on FullCLR, not .NET Core.
-try {
-    $Runtime = [System.Runtime.InteropServices.RuntimeInformation]
-    $OSPlatform = [System.Runtime.InteropServices.OSPlatform]
-
-    $IsCoreCLR = $true
-    $IsLinux = $Runtime::IsOSPlatform($OSPlatform::Linux)
-    $IsOSX = $Runtime::IsOSPlatform($OSPlatform::OSX)
-    $IsWindows = $Runtime::IsOSPlatform($OSPlatform::Windows)
-} catch {
-    # If these are already set, then they're read-only and we're done
+$dotnetCLIChannel = "preview"
+$dotnetCLIRequiredVersion = "2.0.0-preview2-006502"
+
+function Get-EnvironmentInformation
+{
+    $environment = @{}
+    # Use the .NET Core APIs to determine the current platform; if a runtime
+    # exception is thrown, we are on FullCLR, not .NET Core.
     try {
-        $IsCoreCLR = $false
-        $IsLinux = $false
-        $IsOSX = $false
-        $IsWindows = $true
+        $Runtime = [System.Runtime.InteropServices.RuntimeInformation]
+        $OSPlatform = [System.Runtime.InteropServices.OSPlatform]
+
+        $environment += @{'IsCoreCLR' = $true}
+        $environment += @{'IsLinux' = $Runtime::IsOSPlatform($OSPlatform::Linux)}
+        $environment += @{'IsOSX' = $Runtime::IsOSPlatform($OSPlatform::OSX)}
+        $environment += @{'IsWindows' = $Runtime::IsOSPlatform($OSPlatform::Windows)}
+    } catch {
+        # If these are already set, then they're read-only and we're done
+        try {
```

Comment:
I mean the inner `try { ... } catch { }` block in the outter `catch { ... }` block and the comment `# If these are already set, then they're read-only and we're done`